File: android/src/test/java/com/reactnativenavigation/TestApplication.java
package com.reactnativenavigation;

import android.app.Application;

import com.facebook.react.ReactApplication;
import com.facebook.react.ReactHost;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.defaults.DefaultReactHost;

import java.util.Collections;
import java.util.List;


public class TestApplication extends Application implements ReactApplication {
    private final ReactNativeHost host = new ReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
            return true;
        }

        @Override
        protected List<ReactPackage> getPackages() {
            return Collections.emptyList();
        }
    };

    @Override
    public void onCreate() {
        super.onCreate();
        setTheme(androidx.appcompat.R.style.Theme_AppCompat);
    }

    @Override
    public ReactNativeHost getReactNativeHost() {
        return host;
    }

    @Override
    public ReactHost getReactHost() {
        return DefaultReactHost.getDefaultReactHost(this, getReactNativeHost());
    }
}

File: android/src/test/java/com/reactnativenavigation/TestApplication.kt
package com.reactnativenavigation

import android.app.Application
import androidx.appcompat.R
import com.facebook.react.ReactApplication
import com.facebook.react.ReactHost
import com.facebook.react.ReactNativeHost
import com.facebook.react.ReactPackage
import com.facebook.react.devsupport.interfaces.DevSupportManager
import org.mockito.Mockito

class TestApplication : Application(), ReactApplication {
    override val reactNativeHost: ReactNativeHost = object : ReactNativeHost(this@TestApplication) {
        override fun getUseDeveloperSupport(): Boolean {
            return true
        }

        override fun getPackages(): MutableList<ReactPackage> {
            return mutableListOf()
        }
    }

    override fun onCreate() {
        super.onCreate()
        setTheme(R.style.Theme_AppCompat)
    }

    override val reactHost: ReactHost = Mockito.mock(ReactHost::class.java).apply {
        val manager = Mockito.mock(DevSupportManager::class.java)
        Mockito.`when`(manager.devSupportEnabled).thenReturn(true)
        Mockito.`when`(devSupportManager).thenReturn(manager)
    }
}
